chain/client/src/stateless_validation/chunk_endorsement_tracker.rs

```diff
@@ -40,33 +53,84 @@ impl ChunkEndorsementTracker {
         Self {
             epoch_manager,
             chunk_endorsements: SyncLruCache::new(NUM_CHUNKS_IN_CHUNK_ENDORSEMENTS_CACHE),
+            // We can use a different cache size if needed, it does not have to be the same as for `chunk_endorsements`.
+            pending_chunk_endorsements: SyncLruCache::new(NUM_CHUNKS_IN_CHUNK_ENDORSEMENTS_CACHE),
+        }
+    }
+
+    /// Process pending endorsements for the given chunk header.
+    /// It removes these endorsements from the `pending_chunk_endorsements` cache.
+    pub fn process_pending_endorsements(&self, chunk_header: &ShardChunkHeader) {
+        let chunk_hash = &chunk_header.chunk_hash();
+        let chunk_endorsements = {
+            let mut guard = self.pending_chunk_endorsements.lock();
+            guard.pop(chunk_hash)
+        };
+        let chunk_endorsements = match chunk_endorsements {
+            Some(chunk_endorsements) => chunk_endorsements,
+            None => {
+                return;
+            }
+        };
+        tracing::debug!(target: "stateless_validation", ?chunk_hash, "Processing pending chunk endorsements.");
+        for endorsement in chunk_endorsements.values() {
+            let _ = self.process_chunk_endorsement(chunk_header, endorsement.clone())
```

Comment:
it looks weird to use `map_err` if you are not using the result
I suggest changing it to use more conventional `if let Err(error) = self.process_chunk_endorsement(chunk_header, endorsement.clone()) { ... }`